lib/identity_config.rb

```diff
@@ -0,0 +1,62 @@
+class IdentityConfig
+  class << self
+    attr_reader :store
+  end
+
+  CONVERTERS = {
+    uri: proc { |value| URI(value) },
+    string: proc { |value| value.to_s },
+    comma_separated_string_list: proc do |value|
+      value.split(',')
+    end,
+    integer: proc do |value|
+      Integer(value)
+    end,
+    json: proc do |value, options: {}|
+      JSON.parse(value, symbolize_names: options[:symbolize_names])
+    end,
+    boolean: proc do |value|
+      case value
+      when 'true'
+        true
+      when 'false'
+        false
+      else
+        raise 'invalid boolean value'
+      end
+    end,
+  }
+
+  def initialize(read_env)
+    @read_env = read_env
+    @written_env = {}
+  end
+
+  def add(key, type: :string, is_sensitive: false, options: {})
+    value = @read_env[key]
+    raise "#{key} is required but is not present" if value.nil?
+    converted_value = CONVERTERS.fetch(type).call(value, options: options)
+    raise "#{key} is required but is not present" if converted_value.nil?
+
+    @written_env[key] = converted_value
+    @written_env
+  end
+
+  def self.build_store(config_map)
+    config = IdentityConfig.new(config_map)
+    config.add(:aamva_sp_banlist_issuers, type: :json)
+    config.add(:attribute_encryption_key_queue, type: :json)
+    config.add(:aws_kms_regions, type: :json)
+    config.add(:deleted_user_accounts_report_configs, type: :json)
+    config.add(:hmac_fingerprinter_key_queue, type: :json)
+    config.add(:no_sp_campaigns_whitelist, type: :json)
+    config.add(:nonessential_email_banlist, type: :json)
+    config.add(:recurring_jobs_disabled_names, type: :json)
+    config.add(:saml_endpoint_configs, type: :json, options: { symbolize_names: true })
+    config.add(:skip_encryption_allowed_list, type: :json)
+    config.add(:sps_over_quota_limit_notify_email_list, type: :json)
+    final_env = config.add(:valid_authn_contexts, type: :json)
+
+    @store = Struct.new('IdentityConfig', *final_env.keys, keyword_init: true).new(**final_env)
```

Comment:
Let's make sure to redact plz!

```suggestion
    @store = RedactedStruct.new('IdentityConfig', *final_env.keys, keyword_init: true).new(**final_env)
```

And for a follow-up, can we add a `sensitive: ` option to the DSL, default it to `true` and then set it to false on a few things we know are OK to log? ex `aws_kms_regions`, `aamva_sp_banlist_issuers` `recurring_jobs_disabled_names` seem pretty safe to me.

then we can filter for those, like:

```
safe_keys = config.reject { |k, v| v.sensitive? }.keys
...
RedactedStruct.new('IdentityConfig', *final_env.keys, keyword_init: true, allowed_members: safe_keys).new(**final_env)
```
or something

Reply:
I'm open to adding it, but my feeling right now is that since we have such a large config, it won't be super useful to print the entire set, and it may be fine to treat all keys as sensitive. We'd probably end up calling specific keys to check values when needed most of the time anyway?

Reply:
I don't feel that strongly, and I think that since we have a simple hash now, it's probably easiest just to leave it out